src/OrleansRuntime/Streams/QueueBalancer/QueueBalancerBaseClass.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Streams;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams
+{
+    /// <summary>
+    /// Base class for StreamQueueBalancer
+    /// </summary>
+    public abstract class QueueBalancerBaseClass : IStreamQueueBalancer
+    {
+        protected readonly List<IStreamQueueBalanceListener> queueBalanceListeners;
```

Comment:
Does this actually need to be `protected`? Could it be `private` instead? If it's going to be exposed, please document the requirement that it's locked before access